sls_api/endpoints/tools/events.py

```diff
@@ -1272,124 +1273,488 @@ def list_translations(project, translation_id):
+                    .where(tag_table.c.project_id == project_id)
+                    .values(**values)
+                    .returning(*tag_table.c)  # Return the updated row
+                )
+                updated_row = connection.execute(stmt).first()
+
+                if updated_row is None:
+                    # No row was returned: invalid tag_id or project name
+                    return create_error_response("Update failed: no keyword record with the provided 'tag_id' found in project.")
+
+                # If the tag is deleted, also delete any events related to it
+                if "deleted" in values and values["deleted"]:
+                    connection_table = get_table("event_connection")
+                    occurrence_table = get_table("event_occurrence")
+                    event_table = get_table("event")
+
+                    del_upd_value = {
+                        "deleted": 1,
+                        "date_modified": values["date_modified"]
+                    }
+
+                    # Subquery: Get event IDs for tag_id (used in two of the updates)
+                    event_id_subquery = (
+                        select(connection_table.c.event_id)
+                        .where(connection_table.c.tag_id == tag_id)
+                    ).scalar_subquery()
+
+                    # 1. Update event_occurrence where event_id matches subquery
+                    upd_occ_stmt = (
+                        occurrence_table.update()
+                        .where(occurrence_table.c.event_id.in_(event_id_subquery))
+                        .values(**del_upd_value)
+                        .returning(occurrence_table.c.id)
+                    )
+
+                    # 2. Update event where id matches same subquery
+                    upd_event_stmt = (
+                        event_table.update()
+                        .where(event_table.c.id.in_(event_id_subquery))
+                        .values(**del_upd_value)
+                        .returning(event_table.c.id)
+                    )
+
+                    # 3. Update event_connection directly using tag_id filter
+                    upd_conn_stmt = (
+                        connection_table.update()
+                        .where(connection_table.c.tag_id == tag_id)
+                        .values(**del_upd_value)
+                        .returning(connection_table.c.id)
+                    )
+
+                    connection.execute(upd_occ_stmt)
+                    connection.execute(upd_event_stmt)
+                    connection.execute(upd_conn_stmt)
+
+                return create_success_response(
+                    message="Keyword record updated.",
+                    data=updated_row._asdict()
+                )
+
+    except Exception:
+        logger.exception("Exception updating tag.")
+        return create_error_response("Unexpected error: failed to update keyword record.", 500)
```

Comment:
Same here, messages and log lines should use the same wording for consistency and clarity.